### PR TITLE
[IMP] account: add new recurring entry frequencies (bimonthly, 4-months, semi-annual)

### PR DESCRIPTION
**Pull request by rpinset**
## Description

This PR extends the recurring entry functionality in the `account` module by adding three new frequency options:
- **Bimonthly** (every 2 months)
- **Every 4 months** 
- **Semi-annually** (every 6 months)

These additions provide more flexibility for businesses with non-standard billing cycles or specific regulatory requirements.

## Motivation

Currently, Odoo only supports monthly, quarterly, and yearly recurring entries. Many businesses require intermediate frequencies:
- Bimonthly billing is common in utilities and subscription services
- 4-month cycles align with trimester-based academic or fiscal periods
- Semi-annual frequencies are used for insurance premiums and regulatory filings

### Changes made:
1. **Extended `auto_post` field** in `account.move` model to include the three new selection options
2. **Enhanced `_apply_delta_recurring_entries` method** to handle the new period calculations
3. **Added comprehensive unit tests** in `test_recurring_entries.py` to ensure proper functionality

### Technical notes:
- The implementation maintains backward compatibility with existing recurring entries
- New frequencies work seamlessly with the existing auto-post cron job (`_autopost_draft_entries`)
- The method preserves the original day of month when possible (e.g., Jan 31 → May 31 for 4-month frequency)

### Unit tests added:
- `test_apply_delta_recurring_entries_bimonthly`: Validates bimonthly date calculations
- `test_apply_delta_recurring_entries_four_months`: Tests 4-month period handling including month-end edge cases
- `test_apply_delta_recurring_entries_semi_annually`: Verifies semi-annual frequency
- `test_copy_recurring_entries_bimonthly`: Ensures proper copying of recurring entries
- `test_copy_recurring_entries_until_date`: Validates respect of `auto_post_until` date
- `test_standard_frequencies_still_work`: Regression test for existing frequencies

### Manual testing performed:
1. Created journal entries with each new frequency
2. Verified automatic posting via cron job
3. Tested date calculations across year boundaries
4. Validated behavior with different month lengths (28, 29, 30, 31 days)

## Impacted versions

- Target version: master
- Should be forward-ported to: 19.0 18.0 17.0 16.0

## Related

- No related issues or PRs
- This enhancement was inspired by real-world requirements from businesses needing more flexible recurring entry schedules. The implementation is minimal and focused, adding only the necessary code to support the new frequencies without disrupting existing functionality.

### CODE REVIEW

**Comment by rpinset** [comment]
Hello @qdp-odoo @path-odoo @roto-odoo , you are among the latest Odoo contributors. Is there a possibility of a review?

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -4661,8 +4663,29 @@ def _getattrstring(obj, field_name):
...
+        :param period: Period type
+        :return: New date advanced by the period from the date_origin
+        '''
+        deltas = {
+            'monthly': 1,
+            'bimonthly': 2,
+            'quarterly': 3,
+            'four_months': 4,
+            'semi_annually': 6,
+            'yearly': 12
```
Comment (h4818) [inline addons/account/models/account_move.py:4687]:
```suggestion
            'yearly': 12,
```

just a small nitpick

**Review by h4818 — COMMENTED** [review]
Technically this is ok for me, just need to confirm with a PO
@tsb-odoo

**Comment by h4818** [comment]
ok, let's try that. Please target 17.0 (you might need to rebase)

(we will create a task internally to add more flexibility)
